### PR TITLE
LEAN-3577 Migration Implementation

### PR DESCRIPTION
**Pull request by mbartenev-atypon**

### CODE REVIEW

**Review by mnatsheh — COMMENTED** [review]
I suggest we use [semver](https://www.npmjs.com/package/semver) for the versions.

File: package.json [hunk 1]
```diff
@@ -34,9 +34,12 @@
     "@manuscripts/library": "1.3.7",
     "debug": "^4.3.4",
     "jszip": "^3.10.1",
+    "lodash": "^4.17.21",
+    "migration-base": "npm:@manuscripts/transform@2.3.9-LEAN-3577-support-merge",
```
Comment (mnatsheh) [inline package.json:38]: Can this be a dev dependency?

File: src/schema/migration/migrate.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  // now find all versions that we have to migrate that do from version
+}
+
+function ensureVersionAscOrder() {
+  return migrationScripts.sort((a, b) =>
+    semver.eq(a.toVersion, b.toVersion)
+      ? 0
+      : semver.gt(a.toVersion, b.toVersion)
+      ? 1
+      : -1
```
Comment (mnatsheh) [inline src/schema/migration/migrate.ts:71]: There should be a compare function in semver.

Comment (mbartenev-atypon) [reply src/schema/migration/migrate.ts]: Indeed. Replaced.

File: package.json (hunk 1 above)
Comment (mbartenev-atypon) [reply package.json]: Yes, we need it only for running tests. Moved.

**Review by mnatsheh — APPROVED** [review]

**Pull request merged**